File: jupyter2/ps/function/a2b2c2/jl/a2b2c2.jl
### begin my answer

### begin hidden
function f(a, b, c)
    a * a + b * b - c * c
end
### end hidden
### end my answer

function main()
    @assert abs(f(1.0, 2.0, 3.0) - -4.0) <= 1.0e-5
    @assert abs(f(3.0, 4.0, 5.0) -  0.0) <= 1.0e-5
    @assert abs(f(5.0, 6.0, 7.0) - 12.0) <= 1.0e-5
    println("OK")
end

main()

File: jupyter2/ps/function/a2b2c2/rs/a2b2c2.rs
/** begin my answer */

/** begin hidden */
fn f(a : f64, b : f64, c : f64) -> f64 {
    a * a + b * b - c * c
}
/** end hidden */
/** end my answer */

fn main() {
    assert!((f(1.0, 2.0, 3.0) - -4.0).abs() < 1.0e-5);
    assert!((f(3.0, 4.0, 5.0) - -0.0).abs() < 1.0e-5);
    assert!((f(5.0, 6.0, 7.0) - 12.0).abs() < 1.0e-5);
    println!("OK")
}
